contracts/FlowIDTableStaking.cdc

```diff
@@ -1289,56 +1306,25 @@ pub contract FlowIDTableStaking {
 
     /// Gets an array of the node IDs that are proposed and approved for the next epoch
     pub fun getProposedNodeIDs(): [String] {
-        var proposedNodes: [String] = []
 
-        let approvedList = FlowIDTableStaking.getApprovedList()
-        let approvedNodeIDs: {String: Bool} = {}
-        for id in approvedList {
-            approvedNodeIDs[id] = true
-        }
+        let approvedNodeIDs = FlowIDTableStaking.getApprovedList()
 
-        for nodeID in FlowIDTableStaking.getNodeIDs() {
-            let nodeRecord = FlowIDTableStaking.borrowNodeRecord(nodeID)
-            let approved = approvedNodeIDs[nodeID] ?? false
-
-            // To be considered proposed, a node has to have tokens staked + committed equal or above the minimum
-            // Access nodes have a minimum of 0, so they need to be strictly greater than zero to be considered proposed
-            if self.isGreaterThanMinimumForRole(numTokens: self.NodeInfo(nodeID: nodeRecord.id).totalCommittedWithoutDelegators(), role: nodeRecord.role)
-               && approved
-            {
-                proposedNodes.append(nodeID)
+        for nodeID in approvedNodeIDs.keys {
+            if FlowIDTableStaking.nodes[nodeID] == nil {
+                approvedNodeIDs[nodeID] = nil
+                continue
             }
```

Comment:
Would it be possible instead to disallow adding a node ID to the approved list, if that node ID has not already been registered?

The ability to add an un-registered node ID to the approved list is a security issue, if there is an operational oversight, because then anyone can register an approved node ID in between it being approved and the operator registering. 

Currently we mitigate this problem via operational process, where we verify a node ID has been registered before approving it. However, it would remove the latitude for manual error, and potentially simplify the smart contract, if we disallowed this order of operations.

Reply:
yeah, I can add that check to the `setApprovedList` function. Do we want that admin check to fail if one of the approved nodes isn't in the ID table, or do we want it to succeed, but just remove that unregistered node from the approved list?

Reply:
I'm thinking we should probably just fail the transaction since it might be because of a typo or something

Reply:
Agree, I think we should fail the transaction.